styx-e2e-test/src/test/java/com/spotify/styx/e2e_tests/EndToEndTestBase.java

```diff
@@ -96,25 +106,62 @@
 
   int styxApiPort = -1;
 
+  private Iam iam;
   private Datastore datastore;
   private NamespacedKubernetesClient k8s;
 
   private Config schedulerConfig;
 
+  ServiceAccount workflowServiceAccount;
+
   @Before
   public void setUp() throws Exception {
     // Setup namespace
-    log.info("Styx test namespace: {}", namespace);
-    System.setProperty("styx.test.namespace", namespace);
-
+    log.info("Styx test namespace: {}", testNamespace);
```

Comment:
This method does too many things, how about we break it down to smaller methods.

Reply:
For `tearDown` as well.

Reply:
👍